shared-module/bitmaptools/__init__.c

```diff
@@ -252,6 +251,148 @@ void common_hal_bitmaptools_fill_region(displayio_bitmap_t *destination,
+            destination,
+            mp_obj_get_int(tuple_items[0]),
+            mp_obj_get_int(tuple_items[1]));
+
+        // if the current point is not background color ignore it
+        if (current_point_color_value != replaced_color_value) {
+            mp_obj_list_get(fill_area, &list_length, &fill_points);
+            continue;
+        }
+
+        cur_x = mp_obj_int_get_checked(tuple_items[0]);
+        cur_y = mp_obj_int_get_checked(tuple_items[1]);
+
+        if (cur_x < minx) {
+            minx = (int16_t)cur_x;
+        }
+        if (cur_x > maxx) {
+            maxx = (int16_t)cur_x;
+        }
+        if (cur_y < miny) {
+            miny = (int16_t)cur_y;
+        }
+        if (cur_y > maxy) {
+            maxy = (int16_t)cur_y;
+        }
+
+        // fill the current point with fill color
+        displayio_bitmap_write_pixel(
+            destination,
+            mp_obj_get_int(tuple_items[0]),
+            mp_obj_get_int(tuple_items[1]),
+            fill_color_value);
+
+        // add all 4 surrounding points to the list to check
+
+        // ignore points outside of the bitmap
+        if (mp_obj_int_get_checked(tuple_items[1]) - 1 >= 0) {
+            mp_obj_t above_point[] = {
+                tuple_items[0],
+                MP_OBJ_NEW_SMALL_INT(mp_obj_int_get_checked(tuple_items[1]) - 1)
+            };
+            mp_obj_list_append(
+                fill_area,
+                mp_obj_new_tuple(2, above_point));
+        }
+
+        // ignore points outside of the bitmap
+        if (mp_obj_int_get_checked(tuple_items[0]) - 1 >= 0) {
+            mp_obj_t left_point[] = {
+                MP_OBJ_NEW_SMALL_INT(mp_obj_int_get_checked(tuple_items[0]) - 1),
+                tuple_items[1]
+            };
+            mp_obj_list_append(
+                fill_area,
+                mp_obj_new_tuple(2, left_point));
+        }
+
+        // ignore points outside of the bitmap
+        if (mp_obj_int_get_checked(tuple_items[0]) + 1 < destination->width) {
+            mp_obj_t right_point[] = {
+                MP_OBJ_NEW_SMALL_INT(mp_obj_int_get_checked(tuple_items[0]) + 1),
+                tuple_items[1]
+            };
+            mp_obj_list_append(
+                fill_area,
+                mp_obj_new_tuple(2, right_point));
+        }
+
+        // ignore points outside of the bitmap
+        if (mp_obj_int_get_checked(tuple_items[1]) + 1 < destination->height) {
+            mp_obj_t below_point[] = {
+                tuple_items[0],
+                MP_OBJ_NEW_SMALL_INT(mp_obj_int_get_checked(tuple_items[1]) + 1)
+            };
+            mp_obj_list_append(
+                fill_area,
+                mp_obj_new_tuple(2, below_point));
+        }
+
+        mp_obj_list_get(fill_area, &list_length, &fill_points);
+    }
+
+    // set dirty the area so displayio will draw
+    displayio_area_t area = { minx, miny, maxx + 1, maxy + 1};
```

Comment:
Is there a reason the max x/y is +1 that maybe I'm missing?

Reply:
I am not 100% sure about the reason, it seems like `displayio_bitmap_set_dirty_area` might be inclusive on one end and exclusive on the other, I don't see anything documenting that though. And it's not consistent on every fill which definitely seems odd to me.

I did try initially without the `+1`s and some of the fills are not showing the far right and bottom edge lines of pixels being filled. Most of the logic was copied from the `rotozoom` function which does also have the `+1`s in it's dirty area as well.